semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -0,0 +1,507 @@
+                all_input_prior = get_input_state_df(
+                    out_file,
+                    ert.eclConfig().get_gridfile(),
+                    ert.getEnkfFsManager().getEnsembleSize(),
+                    output_path.replace("scalar_", "field_") + "prior/",
+                    fieldparam,
+                )
+                scaler = StandardScaler()
+                scaler.fit(all_input_prior)
+                pca = PCA(0.98).fit(pd.DataFrame(scaler.transform(all_input_prior)))
+                #                print("Number of PCA:", fieldparam, " ", pca.n_components_)
+                pc_fieldprior_df = pd.DataFrame(
+                    data=pca.transform(scaler.transform(all_input_prior))
+                )
+                all_ks = pd.DataFrame()
+                # Get the posterior Field parameters
+                mygrid_ok_short = mygrid_ok[mygrid_ok["KZ"] == 1].copy().reset_index()
+                for key in obs_groups:
+                    all_input_post = get_input_state_df(
+                        out_file,
+                        ert.eclConfig().get_gridfile(),
+                        ert.getEnkfFsManager().getEnsembleSize(),
+                        output_path.replace("scalar_", "field_") + key + "/",
+                        fieldparam,
+                    )
+                    mygrid_ok_short = calc_delta_grid(
+                        all_input_post,
+                        all_input_prior,
+                        mygrid_ok,
+                        key,
+                        mygrid_ok_short,
+                    )
+                    pc_fieldpost_df = pd.DataFrame(
+                        data=pca.transform(scaler.transform(all_input_post))
+                    )
+                    all_ks = calc_ks(
+                        all_ks,
+                        pc_fieldpost_df,
+                        pc_fieldprior_df,
+                        pc_fieldpost_df,
+                        key,
+                    )
+                # add the field max Ks to the scalar Ks matrix
+                ks_data.loc["FIELD_" + fieldparam] = all_ks.max()
+                mygrid_ok_short.to_csv(
+                    output_path.replace("scalar_", "field_")
+                    + "delta_field"
+                    + fieldparam
+                    + ".csv"
+                )
+        # save/export the Ks matrix, active_obs, misfitval and prior data
+        save_to_csv(ks_data, active_obs, misfitval, prior_data, output_path)
+
+
+def initialize_emptydf():
+    ks_data = pd.DataFrame()
+    active_obs = pd.DataFrame()
+    misfitval = pd.DataFrame()
+    return ks_data, active_obs, misfitval
+
+
+def make_update_log_df(update_log_file, key):
+    """ Read update_log file to get active and inactive observations """
+    list_of_files = [
+        os.path.join(update_log_file, f) for f in os.listdir(update_log_file)
+    ]
+    if len(list_of_files) > 1:
+        print("Warning more than one update_log_file.")
```

Comment:
actually in the context of the script there are always only one update log file generated (it is overwritten for each ministep update) and also since it is generated in another folder than the one setup in the ert config it is independent (no risk of confusion/several files). But we can keep the check and I agree return an error if it happens several files are present in the update_log folder (it should not happen).